lib/Backends/CPU/LLVMIRGen.cpp

```diff
@@ -305,7 +306,7 @@ llvm::Value *LLVMIRGen::emitValueAddress(llvm::IRBuilder<> &builder,
                                          const glow::Value *val) {
   assert(allocationsInfo_.allocatedAddress_.count(val) &&
          "Value address was not allocated");
-  auto sizeTTy = builder.getIntNTy(sizeof(size_t) * 8);
+  auto sizeTTy = builder.getIntNTy(TM_->getPointerSizeInBits(0));
```

Comment:
@petecoup Could you please add a short explanation in the commit message and PR header why it is OK to take the number of bits in a pointer as a number of bits in `size_t`.

Reply:
Sure.  In short: when trying to leverage LLVMIRGen in a cross-compiler, we want to emit i32 (i64) types to interface with the corresponding IR types from the cross-compiled libjit for size_t types. This means that these certain types should be "getIntNTy(target sizeof(size_t))", rather than "getIntNTy(host sizeof(size_t))".  So, yes, this change is making the assumption the only supported LLVMIRGen targets have "sizeof(size_t) == sizeof(void*)".  Is that OK (or already the case implicitly)?  If so, I can add something like this in the commit message.  If not, then I'll need to do something else :).

Reply:
The motivation is clear. We definitely want to use the target "getIntNTy(target sizeof(size_t))". 

And most likely we implicitly assume that "sizeof(size_t) == sizeof(void*)", but we do not clearly state it anywhere.

That being said, I'm not sure we want to hard-code this assumption in the code. May be we should add a helper method to `LLVMIRGen` to return the LLVM type representing the target `size_t` integer type? The default implementation could use what you propose, i.e. `builder.getIntNTy(TM_->getPointerSizeInBits(0));` But if this is not the case for a given target, one could easily override this new method to return whatever is appropriate.

Reply:
BTW, we had a lot of headaches in some other backends (e.g. OpenCL) due to the use of size_t. That's why we banned it from the OpenCL analog of `libjit` and use there explicit fixed-width integer types to avoid any mismatches. May be we should do the same for `libjit` to avoid issues with `size_t` size mismatches between the host and cross-target?

Reply:
Sure, makes sense.  I can do something along those lines.